File: src/main/java/com/devops/kanban/entity/Session.java
package com.devops.kanban.entity;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import lombok.Builder;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Session {
    private Long id;
    private Long taskId;
    private Long agentId;
    private SessionStatus status;
    /**
     * For Git projects: path to the isolated Git worktree.
     * For non-Git projects: path to the project's local directory (used directly without worktree).
     */
    private String worktreePath;
    private String branch;
    private String sessionId;      // UUID for WebSocket topic
    private String claudeSessionId; // Claude CLI's native session ID for --resume
    private LocalDateTime startedAt;
    private LocalDateTime lastHeartbeat;
    private LocalDateTime stoppedAt;
    private String output;  // Conversation history/output buffer (deprecated, use messages)
    private String initialPrompt;  // Initial prompt passed to Claude CLI (for frontend filtering)
    private String messages;  // JSON array of chat messages

    // Phase transition fields
    private Boolean phaseCompleteSignal;  // Whether AI has signaled phase completion
    private String targetPhase;            // AI-suggested next phase

    public enum SessionStatus {
        CREATED,    // Session created but not started
        RUNNING,    // Agent process is actively running
        IDLE,       // Agent process is waiting for input
        STOPPED,    // Session was manually stopped
        ERROR       // Session encountered an error
    }
}

File: backend/app/models/session.py
"""Session data model for AI agent execution."""

from datetime import datetime
from enum import Enum
from typing import Optional

from pydantic import BaseModel, Field


class SessionStatus(str, Enum):
    """Session status enumeration."""

    CREATED = "CREATED"
    RUNNING = "RUNNING"
    IDLE = "IDLE"
    STOPPED = "STOPPED"
    COMPLETED = "COMPLETED"
    ERROR = "ERROR"


class AgentType(str, Enum):
    """Agent type enumeration."""

    CLAUDE = "CLAUDE"
    CODEX = "CODEX"
    CURSOR = "CURSOR"
    CUSTOM = "CUSTOM"


class SessionBase(BaseModel):
    """Base session model."""

    task_id: int
    agent_id: int
    status: SessionStatus = Field(default=SessionStatus.CREATED)
    initial_prompt: str = Field(default="")
    branch: Optional[str] = None
    claude_session_id: Optional[str] = None
    worktree_path: Optional[str] = None


class SessionCreate(SessionBase):
    """Session creation model."""

    pass


class SessionUpdate(BaseModel):
    """Session update model."""

    status: Optional[SessionStatus] = None
    initial_prompt: Optional[str] = None
    branch: Optional[str] = None
    claude_session_id: Optional[str] = None
    worktree_path: Optional[str] = None
    output: Optional[str] = None


class Session(SessionBase):
    """Session model with ID and timestamps."""

    id: int
    output: Optional[str] = Field(default="")
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True
